SCClassLibrary/Common/UnitTesting/UnitTest.sc

```diff
@@ -47,30 +48,95 @@ UnitTest {
 	}
 
 	// run a single test in the name format "TestPolyPlayerPool:test_prepareChildrenToBundle"
-	*runTest { | methodName |
-		var class, method, unitTest;
-		# class, method = methodName.split($:);
-		class = class.asSymbol.asClass;
-		method.asSymbol;
-		method = class.findMethod(method.asSymbol);
-		if(method.isNil) { Error("Test method not found "+methodName).throw };
-		class.new.runTestMethod(method);
+	*runTest { | methodString |
+		var method = this.findTestMethodFor(methodString);
+		if(method.isNil) {
+			warn("*** Test method not found "+ method.class ++ ":" ++ method.name)
+		} {
+			method.ownerClass.new.runTestMethod(method)
+		}
 	}
 
 	// run a single test method of this class
 	runTestMethod { | method |
-		var function;
+		var start = Main.elapsedTime, runTime;
 		("RUNNING UNIT TEST" + this.class.name ++ ":" ++ method.name).inform;
 		this.class.forkIfNeeded {
 			this.setUp;
 			currentMethod = method;
 			this.perform(method.name);
 			this.tearDown;
-			this.class.report;
+			runTime = Main.elapsedTime - start;
+			this.class.report(runTime);
 			nil
 		}
 	}
 
+	// run a list of testItems which can be UnitTest classes or method names
+	// in the format "TestPolyPlayerPool:test_prepareChildrenToBundle"
+
+	*runTests { |testItems, postTimes = false|
+		^this.forkIfNeeded {
+			var start = Main.elapsedTime;
+			this.reset;
+			testItems = testItems ?? { this.allSubclasses };
+			testItems.do ({ |testItem|
+				if (testItem.isKindOf(Class)) {
+					// assume this is a UnitTest:
+					if (testItem.isKindOf(Meta_UnitTest)) {
+						testItem.run(false,false);
+					} {
+						"%: testItem % is not a unit test subclass."
+						.format(testItem.cs).warn;
+					}
+				} {
+					if (testItem.isKindOf(String)) {
+						try { UnitTest.runTest(testItem) } {
+							"%: testItem % is not a unit test method string."
+							.format(testItem.cs).warn;
+						}
+					}
+				};
+				0.1.wait;
+			});
+			this.report(Main.elapsedTime - start);
+		}
+	}
+
+	// could be moved to e.g. Method.find(methodString):
+	*findMethodFor { |methodString|
+
+		var className, methodName, class, method;
+		# class, methodName = methodString.split($:);
+		class = class.asSymbol.asClass;
+		if (class.notNil) {
+			method = class.findMethod(methodName.asSymbol);
+		};
+
+		if (method.isNil) {
+			"no method found for: %\n".postf(methodString.cs);
+		};
+		^method
+	}
+
+	// could be moved to e.g. Method.isTest
+	*isTestMethod { |method|
+		^method.ownerClass.superclasses.includes(UnitTest) and: {
```

Comment:
here is another one you can replace by`method.ownerClass.isKindOf(Meta_UnitTest)`